File: packages/sdk/dotnet/Modules/ThreeDApi.cs
using System.Text.Json;

namespace Ihui.AI;

/// <summary>
/// 3D 模型生成模块。
/// </summary>
/// <remarks>
/// 端点(1 个):
/// <list type="bullet">
///   <item>POST /v1/3d/generations</item>
/// </list>
/// </remarks>
public sealed class ThreeDApi
{
    private readonly BaseClient _client;

    internal ThreeDApi(BaseClient client) => _client = client;

    /// <summary>POST /v1/3d/generations(3D 模型生成)。</summary>
    /// <param name="req">请求体(prompt / model / format 等)</param>
    /// <param name="cancellationToken">取消令牌</param>
    /// <returns>生成结果(JsonElement,含 modelUrl / taskId)</returns>
    public Task<JsonElement?> GenerationsAsync(object req, CancellationToken cancellationToken = default)
        => _client.RequestAsync<JsonElement>("POST", "/3d/generations", req, cancellationToken);
}

File: packages/sdk/java/src/main/java/com/ihui/ai/sdk/module/ThreeDApi.java
package com.ihui.ai.sdk.module;

import com.fasterxml.jackson.databind.JsonNode;
import com.ihui.ai.sdk.BaseClient;

import java.util.Map;

/**
 * 3D 模型生成模块。
 *
 * <p>端点(1 个):
 * <ul>
 *   <li>POST /v1/3d/generations</li>
 * </ul>
 */
public final class ThreeDApi {

    private final BaseClient client;

    /**
     * 构造 ThreeDApi。
     *
     * @param client 底层 BaseClient
     */
    public ThreeDApi(BaseClient client) {
        this.client = client;
    }

    /**
     * POST /v1/3d/generations(3D 模型生成)。
     *
     * @param req 请求体(prompt / model / format 等)
     * @return 生成结果(JsonNode,含 modelUrl / taskId)
     */
    public JsonNode generations(Map<String, Object> req) {
        return client.request("POST", "/3d/generations", req, JsonNode.class);
    }
}
